internal/proxy/options.go

```diff
@@ -200,6 +200,14 @@ func (o *Options) Validate() error {
 		}
 	}
 
+	if o.upstreamConfigs != nil {
+		for _, uc := range o.upstreamConfigs {
+			if uc.Timeout > o.TCPWriteTimeout {
+				o.TCPWriteTimeout = uc.Timeout
```

Comment:
What if TCPWriteTimeout is set to 0 to support streaming?

Reply:
I think users will have to implement this at their own peril for the moment. I think we'd have to implement an actually underlying idle tcp connection -- but might be a more involved change.

For the moment, this fix delivers tangible value to users.